scripts/compare_benchmarks.py

```diff
@@ -63,13 +51,85 @@ def calculate_and_format_p_value(old, new):
         return "(run time too short)"
     elif len(old_durations) < min_iterations or len(new_durations) < min_iterations:
         is_significant = False
-        return "(not enough runs)"
-    else:
-        color = "green" if is_significant else "white"
-        return colored("{0:.4f}".format(p_value), color)
 
-
-if len(sys.argv) not in [3, 4]:
+        # in case we cannot decide whether the change is significant due to an insufficient number of measurements, the
+        # add_note_for_insufficient_pvalue_runs flag it set, for which a note is later added to the table output
+        global add_note_for_insufficient_pvalue_runs
+        add_note_for_insufficient_pvalue_runs = True
+        return colored("˅", "yellow", attrs=["bold"])
+    else:
+        if is_significant:
+            return colored(f"{p_value:.4f}", "white")
+        else:
+            return colored(f"{p_value:.4f}", "yellow", attrs=["bold"])
+
+
+def create_context_overview(old_config, new_config, github_format):
+    ignore_difference_for = ["GIT-HASH", "date"]
+
+    old_context_keys = set(old_config["context"].keys())
+    new_context_keys = set(new_config["context"].keys())
+    common_context_keys = old_context_keys & new_context_keys
+
+    table_lines = [["Parameter", sys.argv[1], sys.argv[2]]]
+    for key in sorted(common_context_keys):
+        old_value = old_config["context"][key]
+        new_value = new_config["context"][key]
+        color = "white"
+        marker = " "
+        if old_value != new_value and key not in ignore_difference_for:
+            color = "red"
+            marker = "≠"
+
+        if key == "build_type" and (old_value == "debug" or new_value == "debug"):
+            # Always warn when non-release builds are benchmarked
+            color = "red"
+            marker = "!"
+
+        table_lines.append([colored(marker + key, color), old_value, new_value])
+
+    # print keys that are not present in both contexts
+    for key in sorted(old_context_keys - common_context_keys):
+        value = old_config["context"][key]
+        table_lines.append([colored("≠" + key, "red"), value, "undefined"])
+
+    for key in sorted(new_context_keys - common_context_keys):
+        value = new_config["context"][key]
+        table_lines.append([colored("≠" + key, "red"), "undefined", value])
+
+    table = AsciiTable(table_lines)
+    table.title = "Configuration Overview"
+
+    table_output = str(table.table)
+
+    if github_format:
+        # In case of github formatting, the table's usage of '+' for the tables lines is interpreted as an improvement.
+        # Thus, we prepend an additional space for all lines, except for diverging config lines which get a '-' to mark
+        # the line in red.
```

Comment:
Suggestion: "For GitHub, the output is a fake diff, where a leading '-' marks a deletion and causes the line to be printed in red. We do that for all differing configuration lines. Other lines are prepended with ' '."